app/src/main/java/com/example/itacademyhw/adapter/StateAdapter.kt

```diff
@@ -0,0 +1,29 @@
+package com.example.itacademyhw.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.paging.LoadState
+import androidx.paging.LoadStateAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.example.itacademyhw.ViewHolders.LoadingItemViewHolder
+import com.example.itacademyhw.databinding.ItemLoadingBinding
+
+class StateAdapter : LoadStateAdapter<RecyclerView.ViewHolder>() {
```

Comment:
тут аналогично можно вынести в конструктор context
```suggestion
class StateAdapter(context: Context) : LoadStateAdapter<RecyclerView.ViewHolder>() {
```